clio-client/src/main/scala/org/broadinstitute/clio/client/dispatch/DeliverArraysExecutor.scala

```diff
@@ -0,0 +1,62 @@
+package org.broadinstitute.clio.client.dispatch
+
+import org.broadinstitute.clio.client.commands.DeliverArrays
+import org.broadinstitute.clio.client.util.IoUtil
+import org.broadinstitute.clio.transfer.model.Metadata
+import org.broadinstitute.clio.transfer.model.arrays.{
+  ArraysExtensions,
+  ArraysMetadata
+}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Special-purpose CLP for delivering arrays to FireCloud workspaces.
+  *
+  * Wraps the move CLP with extra IO / upsert logic:
+  *
+  *   1. Writes the arrays md5 value to file at the target path
+  *   2. Records the workspace name in the metadata for the delivered arrays
+  */
+class DeliverArraysExecutor(deliverCommand: DeliverArrays)
+    extends DeliverExecutor(deliverCommand) {
+
+  override def customMetadataOperations(
+    metadata: ArraysMetadata,
+    ioUtil: IoUtil
+  )(
+    implicit ec: ExecutionContext
+  ): Future[ArraysMetadata] = {
+    metadata.grnIdat match {
+      case Some(path) => ioUtil.copyGoogleObject(path, deliverCommand.destination)
+      case None =>
+        throw new IllegalStateException(
+          s"Arrays record with key ${deliverCommand.key} is missing its grnIdatPath"
+        )
+    }
+    metadata.redIdat match {
+      case Some(path) => ioUtil.copyGoogleObject(path, deliverCommand.destination)
+      case None =>
+        throw new IllegalStateException(
+          s"Arrays record with key ${deliverCommand.key} is missing its redIdatPath"
+        )
+    }
+
+    val movedGrnIdat = Metadata.findNewPathForMove(
+      metadata.grnIdat.get,
```

Comment:
We don't need to `.get`, since we're already matching. In the cram delivery we did a tuple match which we could replicate:
```scala
(metadata.grnIdat, metadta.redIdat) match {
  case (Some(grn), Some(red)) => // eventually a Future.successful
  case _ => // eventually a Future.error
}
```